exercises/input.js

```diff
@@ -50,7 +60,34 @@ class Input{
 		if it is still null, no range has been set, so don't test it
 			if it is not null, then test the range */
 	test(){
-
+		let output = {
+			result: false
+		};
+		let valueTest = this.inputElement.val();
+		
+		if (this.pattern.test(valueTest)){
+			let rangeTest = this.getRange();
+			if(!rangeTest.min && !rangeTest.max){
```

Comment:
rather than reusing the variable rangeTest, you could use your method.

```
if(! this.getRange().min && !this.getRange().max){
```

Reply:
This might also do some odd things if min and max were both 0.  Not very useful, sure.  but probably better to test against null

Reply:
Because if min and max were 0, and the number was 5, which would be out of range, this would return true